tools/nvqpp/nvq++.in

```diff
@@ -746,7 +746,8 @@ if ${SHOW_VERSION} && [ -z "$SRCS" ] && [ -z "$OBJS" ]; then
 fi
 
 for i in ${SRCS}; do
-	file=$(basename -s .cc -s .cpp $i)
+	file_with_suffix=$(basename $i)
+	file=${file_with_suffix%%.*}
```

Comment:
This is closer, but will fail if the source file has more than one `.` in the name. (It removes everything after the first `.`.)

Maybe let's try writing a simple function to do this instead. Something like this ought to catch the 4 cases precisely.
```bash
function remove_suffix {
  case $1 in
  *.cpp) rv=`basename -s .cpp $i` ;;
  ...
  esac
  $rv
}
```

Reply:
Nice catch.
The following line should only remove the part after the last `.`:
```bash
file=${file_with_suffix%.*}
```

Would you prefer this little modification or the custom function approach?

Reply:
Nice trick. The simple change is fine.

Reply:
Perfect, that line is now updated.